src/Microsoft.ML.ImageAnalytics/ImageLoader.cs

```diff
@@ -201,11 +239,112 @@ protected override Delegate MakeGetter(DataViewRow input, int iinfo, Func<int, b
                                 throw Host.Except($"Failed to load image {src.ToString()}.");
                         }
                     };
+
+                return del;
+            }
+
+            private Delegate MakeGetterVectorDataViewByteType(DataViewRow input, int iinfo, Func<int, bool> activeOutput, out Action disposer)
+            {
+                Contracts.AssertValue(input);
+                Contracts.Assert(0 <= iinfo && iinfo < _parent.ColumnPairs.Length);
+
+                disposer = null;
+                var getSrc = input.GetGetter<ReadOnlyMemory<char>>(input.Schema[ColMapNewToOld[iinfo]]);
+                ReadOnlyMemory<char> src = default;
+                ValueGetter<VBuffer<byte>> del =
+                    (ref VBuffer<byte> dst) =>
+                    {
+                        getSrc(ref src);
+
+                        if (src.Length > 0)
+                        {
+                            string path = src.ToString();
+                            if (!string.IsNullOrWhiteSpace(_parent.ImageFolder))
+                                path = Path.Combine(_parent.ImageFolder, path);
+
+                            int imgSize = LoadDataIntoBuffer(path, ref dst);
+                            if (imgSize < 0)
+                                throw Host.Except($"Failed to load image {src.ToString()}.");
+                        }
+                    };
+
                 return del;
             }
 
+            public static int LoadDataIntoBuffer(string path, ref VBuffer<byte> imgData)
```

Comment:
>int [](start = 26, length = 3)

The only use of the return value of this method is when it is compared to 0. You can instead rename it to `TryLoadDataIntoBuffer` and change the return value to bool. Or, alternatively, you can have the method return void and make it throw the exception instead of its caller.

Reply:
Thanks. Will do that in my next commit.